src/draw/dma2d/lv_draw_dma2d.c

```diff
@@ -0,0 +1,569 @@
+
+    uint32_t fg_color;
+    uint32_t fg_alpha_mode; /* 3 possible modes - 0x0 to 0x2 */
+    uint32_t fg_alpha;
+
+    void * bg_address;
+    uint32_t bg_offset;
+    lv_draw_dma2d_fgbg_cf_t bg_cf;
+
+} lv_draw_dma2d_configuration_t;
+
+typedef struct {
+    void * first_byte;
+    uint32_t width_bytes;
+    uint32_t height;
+    uint32_t stride;
+} lv_draw_dma2d_cache_area_t;
+
+typedef struct {
+    lv_draw_unit_t base_unit;
+    lv_draw_task_t * task_act;
+    lv_draw_dma2d_cache_area_t writing_area;
+#if LV_USE_OS
+    lv_thread_t thread;
+#if LV_USE_DRAW_DMA2D_INTERRUPT
+    lv_thread_sync_t interrupt_signal;
+#else
+    lv_thread_sync_t thread_begin_polling_signal;
+#endif
+#endif
+} lv_draw_dma2d_unit_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static int32_t evaluate_cb(lv_draw_unit_t * draw_unit, lv_draw_task_t * task);
+static int32_t dispatch_cb(lv_draw_unit_t * draw_unit, lv_layer_t * layer);
+static int32_t delete_cb(lv_draw_unit_t * draw_unit);
+static void opaque_fill(void * first_pixel, int32_t w, int32_t h, int32_t stride, lv_color_t color,
+                        lv_color_format_t cf, lv_draw_dma2d_unit_t * u);
+static void fill(void * first_pixel, int32_t w, int32_t h, int32_t stride, lv_color_t color, lv_color_format_t cf,
+                 lv_opa_t opa, lv_draw_dma2d_unit_t * u);
+static void configure_and_start_transfer(const lv_draw_dma2d_configuration_t * conf, lv_draw_dma2d_unit_t * u);
+#if LV_USE_OS
+    static void thread_cb(void * arg);
+#endif
+#if !LV_USE_DRAW_DMA2D_INTERRUPT
+    static void await_transfer_completion(void);
+#endif
+static void post_transfer_tasks(lv_draw_dma2d_unit_t * u);
+static void invalidate_cache(const lv_draw_dma2d_cache_area_t * mem_area);
+static void clean_cache(const lv_draw_dma2d_cache_area_t * mem_area);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+#if LV_USE_OS || LV_USE_DRAW_DMA2D_INTERRUPT
+    static lv_draw_dma2d_unit_t * g_unit;
+#endif
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_draw_dma2d_init(void)
+{
+    lv_draw_dma2d_unit_t * draw_dma2d_unit = lv_draw_create_unit(sizeof(lv_draw_dma2d_unit_t));
+    draw_dma2d_unit->base_unit.evaluate_cb = evaluate_cb;
+    draw_dma2d_unit->base_unit.dispatch_cb = dispatch_cb;
+    draw_dma2d_unit->base_unit.delete_cb = delete_cb;
+
+#if LV_USE_OS || LV_USE_DRAW_DMA2D_INTERRUPT
+    g_unit = draw_dma2d_unit;
+#endif
+
+#if LV_USE_OS
+    g_unit = draw_dma2d_unit;
+
+#if LV_USE_DRAW_DMA2D_INTERRUPT
+    lv_result_t res = lv_thread_sync_init(&draw_dma2d_unit->interrupt_signal);
+    LV_ASSERT(res == LV_RESULT_OK);
+
+    res = lv_thread_init(&draw_dma2d_unit->thread, LV_THREAD_PRIO_HIGH, thread_cb, 2 * 1024, draw_dma2d_unit);
+    LV_ASSERT(res == LV_RESULT_OK);
+#else
+    lv_result_t res = lv_thread_sync_init(&draw_dma2d_unit->thread_begin_polling_signal);
```

Comment:
I suggest following [this pattern](https://github.com/lvgl/lvgl/blob/d7b82c79668d7b31d52f6535360eff15396eade5/src/draw/renesas/dave2d/lv_draw_dave2d.c#L404-L412) as it's already proven to be work in many environments. (Even if it's also not perfect)